Realm/ObjectServerTests/RLMObjectServerTests.mm

```diff
@@ -446,11 +456,30 @@ - (void)testOpenRealmWithNormalCredentials {
 
 /// If client B adds objects to a synced Realm, client A should see those objects.
 - (void)testAddObjects {
-    NSURL *url = REALM_URL();
-    RLMSyncUser *user = [self logInUserForCredentials:[RLMObjectServerTests basicCredentialsWithName:NSStringFromSelector(_cmd)
+    [self verifyAddObjectsWithUsername:NSStringFromSelector(_cmd) relamURL:REALM_URL() simulateReconnection:NO];
+}
+
+- (void)testAddObjectsWithReconnection {
+    [self verifyAddObjectsWithUsername:NSStringFromSelector(_cmd) relamURL:REALM_URL() simulateReconnection:YES];
+}
+
+- (void)verifyAddObjectsWithUsername:(NSString *)username
+                            relamURL:(NSURL *)url
+                simulateReconnection:(BOOL)simulateReconnection {
+    NSLog(@"isParent: %d", self.isParent);
+    RLMSyncUser *user = [self logInUserForCredentials:[RLMObjectServerTests basicCredentialsWithName:username
                                                                                             register:self.isParent]
-                                              server:[RLMObjectServerTests authServerURL]];
-    RLMRealm *realm = [self openRealmForURL:url user:user];
+                                               server:[RLMObjectServerTests authServerURL]];
+    NSLog(@"Login succeeded: %@", user.identity);
+    if (!self.isParent && simulateReconnection) {
+        [self disableNetworking];
```

Comment:
I don't think this will disable networking for just the child process. Perhaps I'm missing something, but there's no process-specific state passed down to the shell script that calls `pfctl`, nor any argument passed to `pfctl` that would allow it to distinguish one process from another. (If I overlooked something, please let me know.)